internal/controller/k8s/k8scluster/cluster_test.go

```diff
@@ -151,7 +151,8 @@ func TestExternalControlPlaneClientObserve(t *testing.T) {
 						State: ionoscloud.PtrString(k8s.ACTIVE),
 					},
 				}, nil, nil)
-				client.EXPECT().GetKubeConfig(context.Background(), "cluster-id").Return("kubeconfig-base64", nil, nil)
+				rets := "{\"apiVersion\":\"v1\",\"kind\":\"Config\",\"preferences\":{},\"current-context\":\"cluster-admin@exampleK8sCluster\",\"clusters\":[{\"cluster\":{\"certificate-authority-data\":\"cadata\",\"server\":\"https://domain.example\"},\"name\":\"exampleK8sCluster\"}],\"contexts\":[{\"context\":{\"cluster\":\"cluster-name\",\"user\":\"cluster-admin\"},\"name\":\"cluster-admin@exampleK8sCluster\"}],\"users\":[{\"name\":\"cluster-admin\",\"user\":{\"token\":\"longtoken\"}}]}"
```

Comment:
```suggestion
				rets := `{"apiVersion":"v1","kind":"Config","preferences":{},"current-context":"cluster-admin@exampleK8sCluster","clusters":[{"cluster":{"certificate-authority-data":"Y2FkYXRh","server":"https://domain.example"},"name":"exampleK8sCluster"}],"contexts":[{"context":{"cluster":"cluster-name","user":"cluster-admin"},"name":"cluster-admin@exampleK8sCluster"}],"users":[{"name":"cluster-admin","user":{"token":"bG9uZ3Rva2Vu"}}]}`
```

Reply:
this should fix the decoding error and the corruptinput check can be removed since we pass base64 string for token and data